service/internal/credentials/vault.go

```diff
@@ -186,6 +187,20 @@ func genProjectAppRole(name string) string {
 	return fmt.Sprintf("%s/%s-%s", vaultAppRolePrefix, vaultProjectPrefix, name)
 }
 
+func (v VaultProvider) CreateToken(name string) (string, string, string, error) {
```

Comment:
Most methods here check if the request is coming from an admin. Should we do the same here?